### PR TITLE
feat(trusted): add cloudinit user-data for vms

### PR DESCRIPTION
**Pull request by smerle33**
as per https://github.com/jenkins-infra/helpdesk/issues/3486

### CODE REVIEW

File: trusted.ci.jenkins.io.tf [hunk 1]
```diff
@@ -131,6 +131,14 @@ resource "azurerm_linux_virtual_machine" "trusted_bounce" {
     public_key = "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAACAQC5K7Ro7jBl5Kc68RdzG6EXHstIBFSxO5Da8SQJSMeCbb4cHTYuBBH8jNsAFcnkN64kEu+YhmlxaWEVEIrPgfGfs13ZL7v9p+Nt76tsz6gnVdAy2zCz607pAWe7p4bBn6T9zdZcBSnvjawO+8t/5ue4ngcfAjanN5OsOgLeD6yqVyP8YTERjW78jvp2TFrIYmgWMI5ES1ln32PQmRZwc1eAOsyGJW/YIBdOxaSkZ41qUvb9b3dCorGuCovpSK2EeNphjLPpVX/NRpVY4YlDqAcTCdLdDrEeVqkiA/VDCYNhudZTDa8f1iHwBE/GEtlKmoO6dxJ5LAkRk3RIVHYrmI6XXSw5l0tHhW5D12MNwzUfDxQEzBpGK5iSfOBt5zJ5OiI9ftnsq/GV7vCXfvMVGDLUC551P5/s/wM70QmHwhlGQNLNeJxRTvd6tL11bof3K+29ivFYUmpU17iVxYOWhkNY86WyngHU6Ux0zaczF3H6H0tpg1Ca/cFO428AVPw/RTJpcAe6OVKq5zwARNApQ/p6fJKUAdXap+PpQGZlQhPLkUbwtFXGTrpX9ePTcdzryCYjgrZouvy4ZMzruJiIbFUH8mRY3xVREVaIsJakruvgw3b14oQgcB4BwYVBBqi62xIvbRzAv7Su9t2jK6OR2z3sM/hLJRqIJ5oILMORa7XqrQ=="
   }
 
+  # Use cloud-init to configure the VM
+  user_data = base64encode(templatefile("./.shared-tools/terraform/cloudinit.tftpl", { hostname = "bounce.trusted.ci.jenkins.io" }))
+  # Force VM recreation when the VPN URL change
```
Comment (dduportal) [inline trusted.ci.jenkins.io.tf:136]:
```suggestion
```

This comment looks like a copy and paste ?

File: trusted.ci.jenkins.io.tf [hunk 2]
```diff
@@ -131,6 +131,14 @@ resource "azurerm_linux_virtual_machine" "trusted_bounce" {
...
+  computer_name = "bounce-trusted-ci-jenkins-io"
```
Comment (dduportal) [inline trusted.ci.jenkins.io.tf:137]:
```suggestion
  computer_name = "bounce.trusted.ci.jenkins.io"
```

Is there a reason not to use the FQDN as hostname?

File: trusted.ci.jenkins.io.tf [hunk 3]
```diff
@@ -178,6 +186,14 @@ resource "azurerm_linux_virtual_machine" "trusted_ci_controller" {
     public_key = "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAACAQC5K7Ro7jBl5Kc68RdzG6EXHstIBFSxO5Da8SQJSMeCbb4cHTYuBBH8jNsAFcnkN64kEu+YhmlxaWEVEIrPgfGfs13ZL7v9p+Nt76tsz6gnVdAy2zCz607pAWe7p4bBn6T9zdZcBSnvjawO+8t/5ue4ngcfAjanN5OsOgLeD6yqVyP8YTERjW78jvp2TFrIYmgWMI5ES1ln32PQmRZwc1eAOsyGJW/YIBdOxaSkZ41qUvb9b3dCorGuCovpSK2EeNphjLPpVX/NRpVY4YlDqAcTCdLdDrEeVqkiA/VDCYNhudZTDa8f1iHwBE/GEtlKmoO6dxJ5LAkRk3RIVHYrmI6XXSw5l0tHhW5D12MNwzUfDxQEzBpGK5iSfOBt5zJ5OiI9ftnsq/GV7vCXfvMVGDLUC551P5/s/wM70QmHwhlGQNLNeJxRTvd6tL11bof3K+29ivFYUmpU17iVxYOWhkNY86WyngHU6Ux0zaczF3H6H0tpg1Ca/cFO428AVPw/RTJpcAe6OVKq5zwARNApQ/p6fJKUAdXap+PpQGZlQhPLkUbwtFXGTrpX9ePTcdzryCYjgrZouvy4ZMzruJiIbFUH8mRY3xVREVaIsJakruvgw3b14oQgcB4BwYVBBqi62xIvbRzAv7Su9t2jK6OR2z3sM/hLJRqIJ5oILMORa7XqrQ=="
   }
 
+  # Use cloud-init to configure the VM
+  user_data = base64encode(templatefile("./.shared-tools/terraform/cloudinit.tftpl", { hostname = "trusted.ci.jenkins.io" }))
+  # Force VM recreation when the VPN URL change
+  computer_name = "trusted-ci-jenkins-io"
```
Comment (dduportal) [inline trusted.ci.jenkins.io.tf:192]:
```suggestion
  computer_name = "trusted.ci.jenkins.io"
```

File: trusted.ci.jenkins.io.tf [hunk 4]
```diff
@@ -251,6 +267,14 @@ resource "azurerm_linux_virtual_machine" "trusted_permanent_agent" {
     public_key = "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAACAQC5K7Ro7jBl5Kc68RdzG6EXHstIBFSxO5Da8SQJSMeCbb4cHTYuBBH8jNsAFcnkN64kEu+YhmlxaWEVEIrPgfGfs13ZL7v9p+Nt76tsz6gnVdAy2zCz607pAWe7p4bBn6T9zdZcBSnvjawO+8t/5ue4ngcfAjanN5OsOgLeD6yqVyP8YTERjW78jvp2TFrIYmgWMI5ES1ln32PQmRZwc1eAOsyGJW/YIBdOxaSkZ41qUvb9b3dCorGuCovpSK2EeNphjLPpVX/NRpVY4YlDqAcTCdLdDrEeVqkiA/VDCYNhudZTDa8f1iHwBE/GEtlKmoO6dxJ5LAkRk3RIVHYrmI6XXSw5l0tHhW5D12MNwzUfDxQEzBpGK5iSfOBt5zJ5OiI9ftnsq/GV7vCXfvMVGDLUC551P5/s/wM70QmHwhlGQNLNeJxRTvd6tL11bof3K+29ivFYUmpU17iVxYOWhkNY86WyngHU6Ux0zaczF3H6H0tpg1Ca/cFO428AVPw/RTJpcAe6OVKq5zwARNApQ/p6fJKUAdXap+PpQGZlQhPLkUbwtFXGTrpX9ePTcdzryCYjgrZouvy4ZMzruJiIbFUH8mRY3xVREVaIsJakruvgw3b14oQgcB4BwYVBBqi62xIvbRzAv7Su9t2jK6OR2z3sM/hLJRqIJ5oILMORa7XqrQ== smerle@MacBook-Pro-de-Stephane.local"
   }
 
+  # Use cloud-init to configure the VM
+  user_data = base64encode(templatefile("./.shared-tools/terraform/cloudinit.tftpl", { hostname = "agent.trusted.ci.jenkins.io" }))
+  # Force VM recreation when the VPN URL change
+  computer_name = "agent-trusted-ci-jenkins-io"
```
Comment (dduportal) [inline trusted.ci.jenkins.io.tf:273]:
```suggestion
  computer_name = "agent.trusted.ci.jenkins.io"
```

**Review by dduportal — COMMENTED** [review]
FIrst set of review and suggestions

File: trusted.ci.jenkins.io.tf (hunk 2 above)
Comment (dduportal) [reply trusted.ci.jenkins.io.tf:136]: Ping @lemeurherve do you remember why you set the VM hostname with carrets instead of using FQDN with dots?

Comment (lemeurherve) [reply trusted.ci.jenkins.io.tf:140]: There must have been a reason but I don't remember which one, and the terraform provider doc doesn't indicate any constraint on this field.

**Pull request merged**